File: scripts/init-db.js
require('dotenv').config();
const { Pool } = require('pg');
const bcrypt = require('bcryptjs');

const pool = new Pool({
    connectionString: process.env.DATABASE_URL,
    ssl: process.env.NODE_ENV === 'production' ? { rejectUnauthorized: false } : false
});

async function initializeDatabase() {
    try {
        // Create admin user
        const adminPassword = await bcrypt.hash('admin123', 10);
        await pool.query(`
            INSERT INTO users (email, password_hash, role)
            VALUES ($1, $2, $3)
            ON CONFLICT (email) DO NOTHING
        `, ['admin@cybershielduk.com', adminPassword, 'admin']);

        console.log('Database initialized successfully');
    } catch (error) {
        console.error('Error initializing database:', error);
    } finally {
        await pool.end();
    }
}

initializeDatabase(); 

File: scripts/init_db.py
import os
import psycopg2
import bcrypt
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

def init_database():
    try:
        # Connect to database
        conn = psycopg2.connect(
            os.getenv('DATABASE_URL'),
            sslmode='require' if os.getenv('NODE_ENV') == 'production' else 'disable'
        )
        cur = conn.cursor()

        # Read and execute schema.sql
        with open('schema.sql', 'r') as f:
            schema = f.read()
            cur.execute(schema)

        # Create admin user
        admin_password = bcrypt.hashpw('admin123'.encode('utf-8'), bcrypt.gensalt())
        cur.execute("""
            INSERT INTO users (email, password_hash, role)
            VALUES (%s, %s, %s)
            ON CONFLICT (email) DO NOTHING
        """, ('admin@cybershielduk.com', admin_password.decode('utf-8'), 'admin'))

        conn.commit()
        print("Database initialized successfully")

    except Exception as e:
        print(f"Error initializing database: {str(e)}")
    finally:
        cur.close()
        conn.close()

if __name__ == '__main__':
    init_database() 